test/Date.test.js

```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import expect from 'expect';
+import { shallow } from 'enzyme';
+import Date from '../js/components/Message/Types/JSObject/Date';
+import data from "./data/Date.js";
+
+describe('Date', () => {
+
+  data.forEach((testObj, testName) => {
+    it(`should render a ${testName}`, () => {
+      const message = testObj.responsePacket.result;
+
+      const rendered = shallow(<Date message={message} />);
+      expect(rendered.text()).toContain(testObj.consoleOutput);
```

Comment:
Maybe at some stage we will want to enhance this to check more of a DOM representation in the consoleOutput property.  As an example with table output, checking if text is contained might not be giving enough info, and even for a normal output the assertion would still pass if extra stuff was added to the message as long as the text hasn't been removed.